editoast/src/views/scenario.rs

```diff
@@ -142,18 +187,26 @@ async fn create(
     let scenario: Scenario = data.into_inner().into_scenario(study_id, timetable_id);
     let scenario = scenario.create(db_pool.clone()).await?;
 
-    // Update study last_modification field
-    let study = study.update_last_modified(db_pool.clone()).await?;
-    study.expect("Study should exist");
-
-    // Update project last_modification field
-    let project = project.update_last_modified(db_pool.clone()).await?;
-    project.expect("Project should exist");
-
-    // Return study with list of scenarios
-    let scenarios_with_trains = scenario.with_details(db_pool).await?;
-
-    Ok(Json(scenarios_with_trains))
+    let scenarios_with_details = conn
+        .transaction::<_, InternalError, _>(|conn| {
+            async {
+                // Update study last_modification field
+                let _study = study.clone().update_last_modified_conn(conn).await?;
+                study = _study.expect("Study should exist");
+
+                // Update project last_modification field
+                let _project = project.clone().update_last_modified_conn(conn).await?;
+                project = _project.expect("Project should exist");
+
+                // Return study with list of scenarios
+                scenario.with_details_conn(conn).await
```

Comment:
Likewise about variable names + unnecessary mutability

```suggestion
    let (project, study, scenarios_with_details) = conn
        .transaction::<_, InternalError, _>(|conn| {
            async {
                // Update study last_modification field
                let study = study.clone().update_last_modified_conn(conn).await?.expect("Study should exist");

                // Update project last_modification field
                let project = project.clone().update_last_modified_conn(conn).await?.expect("Project should exist");

                (project, study, scenario.with_details_conn(conn).await)
```